src/Razor/test/Microsoft.AspNetCore.Razor.Language.Test/DefaultRazorProjectFileSystemTest.cs

```diff
@@ -41,6 +41,25 @@ public void NormalizeAndEnsureValidPath_NormalizesToAbsolutePath()
             Assert.Equal("C:/some/test/path/root/file.cshtml", absolutePath);
         }
 
+        [Fact]
+        public void NormalizeAndEnsureValidPath_FileFromNetworkShare_NormalizesToAbsolutePath()
+        {
+            // Arrange
+            var fileSystem = new TestRazorProjectFileSystem("//some/network/share/root");
+
+            // Act 1
+            var absolutePath = fileSystem.NormalizeAndEnsureValidPath("\\\\some\\network\\share\\root\\file.cshtml");
+
+            // Assert 1
+            Assert.Equal("//some/network/share/root/file.cshtml", absolutePath);
+
+            // Act 2
+            absolutePath = fileSystem.NormalizeAndEnsureValidPath("//some/network/share/root/file.cshtml");
+
+            // Assert 2
+            Assert.Equal("//some/network/share/root/file.cshtml", absolutePath);
```

Comment:
These are two tests not one. This is a filthy habit

Reply:
Done